contributing guidelines review (#457)
ipfs/community: CONTRIBUTING_JS.md

#### Documentation

> `TBW`

Reviewer: Does this section want removing for the time being?  What does `TBW` mean?

Author: To be written. Right now, AFAIK we don't use nor rely on documentation.js for anything.

### Commits

> We have very precise rules over how our git commit messages can be formatted. This leads to more readable messages that are easy to follow when looking through the project history.  But also, we use the git commit messages to generate the change log.

Reviewer 2: ```suggestion
We have guidelines for how our git commit messages should be formatted. This leads to more readable messages that are easy to follow when looking through the project history.  But also, we use the git commit messages to generate the change log.
```

Reasoning: https://github.com/ipfs/community/pull/383#discussion_r245245383

The commit message formatting can be added using a typical git workflow or through the use of a CLI wizard ([Commitizen](https://github.com/commitizen/cz-cli)).

- **Type** - Must be one of the following:
  - **feat**: A new feature
  - **fix**: A bug fix
  - **docs**: Documentation only changes
  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
  - **refactor**: A code change that neither fixes a bug nor adds a feature
  - **perf**: A code change that improves performance
  - **test**: Adding missing tests
  - **chore**: Changes to the build process or auxiliary tools and libraries such as documentation generation
- **Scope** - The scope could be anything specifying the place of the commit change. For example `api`, `cli`, etc...

> - **Breaking Changes** - Should be part of the commit message. Start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.

Reviewer 2: ```suggestion
- **Breaking Changes** - Should be identified at the end of commit message. Start with the words `BREAKING CHANGE:` on a new line followed by a space or two new lines. The rest of the commit message is then used to describe in detail what was broken and the migration path (if there is one).
```

> Examples:

Reviewer: Could you add an example with a breaking change message please?

Reviewer 2: @achingbrain there is already one below ;)

Reviewer: There's an example, but it's just a regular feature commit?

Author: See

<img width="801" alt="image" src="https://user-images.githubusercontent.com/1211152/75784618-c6b2b480-5d62-11ea-9fee-13e370e79566.png">
